pipeline/testing/sast/ledger.py

```
@@ -0,0 +1,253 @@
+        raise LedgerError("条目 %d disposition=false_positive 必须带 reason" % index)
+    if disposition == "fixed" and not SHA_RE.match(str(entry.get("resolved_sha", ""))):
+        raise LedgerError("条目 %d disposition=fixed 必须带 40 位 hex resolved_sha" % index)
+    if not DATE_RE.match(str(entry.get("date", ""))):
+        raise LedgerError("条目 %d date 非法: %r（需 YYYY-MM-DD）" % (index, entry.get("date")))
+    extra = set(entry) - set(FIELD_ORDER)
+    if extra:
+        raise LedgerError("条目 %d 含未知字段: %s" % (index, sorted(extra)))
+
+
+def verify_ledger(path):
+    """全量校验：结构 + 逐条 disposition 规则 + sha256 链。返回 (entries, ok, detail)。"""
```

Ruff (0.16.2)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)